compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -2022,7 +2022,8 @@ class Typer extends Namer
   private def typedQuotePattern(quoted: untpd.Tree, pt: Type, quoteSpan: Span)(implicit ctx: Context): Tree = {
     val exprPt = pt.baseType(defn.QuotedExprClass)
     val quotedPt = if (exprPt.exists) exprPt.argTypesHi.head else defn.AnyType
```

Comment:
I don't see why `.head` would be the right bound in all cases. I propose to do the safe thing instead:
```
quotedPt = exprPt.argInfos.headOption match {
  case Some(argPt: ValueType) => // excludes TypeBounds
    argPt
  case _ =>
    defn.AnyType
}
```

Reply:
Updated